src/views/ThreeWProjectForm/index.tsx

```diff
@@ -0,0 +1,1004 @@
+import NonFieldError from '#components/NonFieldError';
+import MultiSelectInput from '#components/MultiSelectInput';
+import useAlert from '#hooks/useAlert';
+import Switch from '#components/parked/Switch';
+import RichTextArea from '#components/parked/RichTextArea';
+import RadioInput from '#components/RadioInput';
+import Checkbox from '#components/Checkbox';
+import TextOutput from '#components/TextOutput';
+import CountrySearchSelectInput from '#components/CountrySearchSelectInput';
+import DistrictMultiSelectInput, { DistrictItem } from '#components/DistrictMultiSelectInput';
+import EventSelectInput, { EventItem } from '#components/EventSelectInput';
+import ServerEnumsContext from '#contexts/server-enums';
+import {
+    PROJECT_STATUS_COMPLETED,
+    PROJECT_STATUS_ONGOING,
+    PROJECT_STATUS_PLANNED,
+    OPERATION_TYPE_EMERGENCY,
+    PROGRAMME_TYPE_DOMESTIC,
+    PROGRAMME_TYPE_MULTILATERAL,
+    OPERATION_TYPE_PROGRAMME,
+} from '#utils/constants';
+import {
+    useRequest,
+    useLazyRequest,
+} from '#utils/restRequest';
+
+import useTranslation from '#hooks/useTranslation';
+import { injectClientId } from '#utils/common';
+
+import schema, { ProjectResponseBody, FormType, AnnualSplit } from './schema';
+import AnnualSplitInput from './AnnualSplitInput';
+
+import styles from './styles.module.css';
+import i18n from './i18n.json';
+
+const defaultFormValues: PartialForm<FormType> = {
+    project_districts: [],
+    secondary_sectors: [],
+    visibility: 'public',
+};
+
+// eslint-disable-next-line import/prefer-default-export
+export function Component() {
+    const strings = useTranslation(i18n);
+    const alert = useAlert();
+
+    const { projectId } = useParams<{ projectId: string }>();
+    const {
+        threeWProjectEdit: threeWProjectEditRoute,
+    } = useContext(RouteContext);
+
+    const {
+        value,
+        setValue,
+        setFieldValue,
+        error: formError,
+        setError: onErrorSet,
+        validate,
+    } = useForm(schema, { value: defaultFormValues });
+
+    const navigate = useNavigate();
+
+    const [districtOptions, setDistrictOptions] = useState<
+        DistrictItem[] | undefined | null
+    >([]);
+
+    const [events, setEvents] = useState<
+        EventItem[] | undefined | null
+    >([]);
+
+    const { pending: pendingProjectDetails } = useRequest({
+        skip: isFalsyString(projectId),
+        url: '/api/v2/project/{id}/',
+        pathVariables: isTruthyString(projectId) ? {
+            id: Number(projectId),
+        } : undefined,
+        onSuccess: (response) => {
+            setDistrictOptions(response.project_districts_detail);
+            setEvents([{
+                ...response.event_detail,
+                // FIXME: event dtype id is a must inside event mini but
+                // its not defined under event_detail of this response
+                dtype: { id: response.event_detail?.dtype } as EventItem['dtype'],
```

Comment:
Should this type error be fixed on the server?

Reply:
It doesn't cause any issues, but I think it should